std/experimental/ndslice/computation.d

```diff
@@ -0,0 +1,237 @@
+    $(HTTP en.wikipedia.org/wiki/Map_(higher-order_function), Map (higher-order function))
++/
+template ndmap(fun...)
+    if (fun.length >= 1)
+{
+    auto ndmap(size_t N, Range)(auto ref Slice!(N, Range) slice)
+    {
+        mixin (SetMapFun);
+        alias M = Map!(SlicePtr!Range, _fun);
+        alias R = Slice!(N, M);
+        with(slice) return R(_lengths, _strides, M(_ptr));
+    }
+
+    auto ndmap(size_t N, Range : Slice!(NI, RangeI), size_t NI, RangeI)
+        (auto ref Slice!(N, Range) slice)
+    {
+        mixin (SetMapFun);
+        alias Ptr = Pack!(NI - 1, RangeI);
+        alias M = Map!(Ptr, _fun);
+        alias R = Slice!(N, M);
+        return R(slice._lengths[0 .. N], slice._strides[0 .. N],
+            M(Ptr(slice._lengths[N .. $], slice._strides[N .. $], slice._ptr)));
+    }
+
+    // may be unified with std.algorithms.iteration.map
+    // after ndslice be removed from the Mir library.
+    private enum SetMapFun =
+    q{
+        static if (fun.length > 1)
+        {
+            import std.functional : adjoin, unaryFun;
+
+            alias _funs = staticMap!(unaryFun, fun);
+            alias _fun = adjoin!_funs;
+
+            // Once DMD issue #5710 is fixed, this validation loop can be moved into a template.
+            foreach (f; _funs)
+            {
+                static assert(!is(typeof(f(RE.init)) == void),
+                    "Mapping function(s) must not return void: " ~ _funs.stringof);
+            }
+        }
+        else
+        {
+            import std.functional : unaryFun;
+
+            alias _fun = unaryFun!fun;
+            alias _funs = AliasSeq!(_fun);
+
+            // Do the validation separately for single parameters due to DMD issue #15777.
+            static assert(!is(typeof(_fun(RE.init)) == void),
+                "Mapping function(s) must not return void: " ~ _funs.stringof);
+        }
+    };
+}
+
+///
+pure nothrow unittest
+{
+    import std.experimental.ndslice.selection : iotaSlice;
+
+    auto s = iotaSlice(2, 3).ndmap!(a => a * 3);
+    assert(s == [[0, 3, 6], [9, 12, 15]]);
+}
+
+pure nothrow unittest
+{
+    import std.experimental.ndslice.selection : iotaSlice;
+
+    assert(iotaSlice(2, 3).slice.ndmap!"a * 2" == [[0, 2, 4], [6, 8, 10]]);
+}
+
+/// Packed slices
+pure nothrow unittest
+{
+    import std.algorithm.iteration : sum;
+    import std.experimental.ndslice.selection : iotaSlice, windows, byElement;
+
+    // iotaSlice(2, 3)
+    // -----
+    // 0 1 2
+    // 3 4 5
+    // -----
+    auto s = iotaSlice(2, 3)
+        .windows(2, 2)
+        .ndmap!(a => a.byElement.sum);
+
+    assert(s == [[0+1+3+4, 1+2+4+5]]);
+}
+
+pure nothrow unittest
+{
+    import std.algorithm.iteration : sum;
+    import std.experimental.ndslice.selection : iotaSlice, windows, byElement;
+
+    auto s = iotaSlice(2, 3)
+        .slice
+        .windows(2, 2)
+        .ndmap!(a => a.byElement.sum);
+
+    assert(s == [[0+1+3+4, 1+2+4+5]]);
```

Comment:
It is not example, but unittest for allocated data (the example if for generated data).